src/ralph/models/conversion_set.py

```diff
@@ -0,0 +1,56 @@
+"""Base conversion set definition"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Callable, Union
+
+from pydantic import BaseModel
+
+from ralph.defaults import MODEL_PATH_SEPARATOR
+from ralph.utils import get_dict_value_from_path
+
+
+@dataclass(frozen=True)
+class ConversionItem:
+    """Represents a conversion set item."""
+
+    dest: tuple[str]
+    src: Union[tuple[str], None]
+    transformers: tuple[Callable[[Any], Any]]
+
+    def __init__(self, dest: str, src=None, transformers=lambda _: _):
+        object.__setattr__(self, "dest", tuple(dest.split(MODEL_PATH_SEPARATOR)))
+        src = tuple(src.split(MODEL_PATH_SEPARATOR)) if src else None
+        object.__setattr__(self, "src", src)
+        transformers = tuple([transformers]) if callable(transformers) else transformers
+        object.__setattr__(self, "transformers", transformers)
```

Comment:
We have to admit that this frozen dataclass API is ugly :sweat_smile:

Reply:
Yes Indeed :smile: we could use regular dataclasses if we would put them in a `conversion_list` instead of a `conversion_ set`. WDYT ?

Reply:
The question is: using a set is smart but do we need advanced set features (union, etc.) or a list would be sufficient?

Reply:
For now we only use `union` to combine the `base` conversion_set with `server_event` or `page_closed` conversion_set, which is almost the same as appending to a list.
But, Indeed it could be useful in future to easily construct new conversion_sets by adding/removing/intersecting etc.)